dpnp/dpnp_iface_mathematical.py

```diff
@@ -908,6 +910,17 @@ def gradient(x1, *varargs, **kwargs):
     return call_origin(numpy.gradient, x1, *varargs, **kwargs)
 
 
+def imag(val):
+    """
+    Return the imaginary part of the complex argument.
+
+    For full documentation refer to :obj:`numpy.imag`.
+
+    """
+
+    return dpnp.ndarray(val.get_array().shape, buffer=val.get_array().imag)
```

Comment:
Better to wrap `val` by `dpnp_descriptor`, then call method `get_array()` of `dpnp_descriptor` object. The method will check type of input object and be able to raise understandable exception.

Reply:
done